radix-engine/src/engine/errors.rs

```diff
@@ -67,7 +70,8 @@ pub enum KernelError {
     RENodeNotInTrack,
 
     // Substate
-    Reentrancy(SubstateId),
+    NodeToSubstateFailure(NodeToSubstateFailure),
```

Comment:
Since Node to Substate mapping is static/deterministic feels like this error should be an unbroken invariant and cause a panic if it ever occurs?

Reply:
I'm happy to either A) make this an invariant - every node must be persistable at the end of a transaction OR B) make `Track` more dynamic and allow nodes to define whether it's ready for persistence.  

Imagine, we have:
```
struct Node {
   substate: A,
   transient: B,
}
```
Then we can define interesting rules, like the transaction must abort if `transient > 3`.

If it's okay for you, I can leave a TODO to explore both options as a follow-up.

Reply:
Mmm, I think there's a better third option where persisted vs. transient is just a property of the substate (or whatever name we want to put to it) but yeah let's just leave this as a TODO